File: python/Level 2/위장.py
from itertools import combinations


def solution(clothes):
    answer = 1
    cnt = {}
    for _, type_ in clothes:
        if type_ in cnt:
            cnt[type_] += 1
        else:
            cnt[type_] = 1

    l = list(cnt.values())
    for i in l:
        answer *= i + 1

    return answer - 1

File: golang/Level 2/위장.go
func solution(clothes [][]string) int {
    answer := 1
    d := make(map[string]int)
    for _, cloth := range clothes {
        key := cloth[1]
        
        ok := false
        for t, _ := range d {
            if t == key {
                d[key]++
                ok = true
                break
            } 
        }
        if !ok {
            d[key] = 1
        }
    }
    
    for _, val := range d {
        answer *= val + 1
    }
    return answer - 1
}
